index.js

```diff
@@ -117,6 +121,39 @@ f1.prototype = {
    */
   targets: function( targets ) {
 
+    var instance = this;
+
+    if (targets) {
+      _.forEach(targets, function (el, elementId) {
+        var elementData = {
+          id: elementId,
+          get states() {
```

Comment:
Defining getters and setters like this are es6 features. For now lets try to keep `f1` as es5